client/ayon_houdini/plugins/create/create_generic.py

```diff
@@ -0,0 +1,751 @@
+        num_components=1
+    )
+
+
+def set_values(node: "hou.OpNode", values: dict):
+    """Set parm values only if both the raw value (e.g. expression) or the
+    evaluated value differ. This way we preserve expressions if they happen
+    to evaluate to a matching value.
+
+    Parms must exist on the node already.
+
+    """
+    for key, value in values.items():
+
+        parm = node.parm(key)
+
+        try:
+            unexpanded_value = parm.unexpandedString()
+            if unexpanded_value == value:
+                # Allow matching expressions
+                continue
+        except hou.OperationFailed:
+            pass
+
+        if parm.rawValue() == value:
+            continue
+
+        if parm.eval() == value:
+            # Needs no change
+            continue
+
+        # TODO: Set complex data types as `JSON:`
+        parm.set(value)
+
+
+@dataclasses.dataclass
+class NodeTypeProductTypes:
+    """Product type settings for a node type.
+
+    Define the available product types the user can set on a ROP based on
+    node type.
+
+    When 'strict' an enum attribute is created and the user can not type a
+    custom product type, otherwise a string attribute is
+    created with a menu right hand side to help pick a type but allow custom
+    types.
+    """
+    product_types: List[str]
+    default: Optional[str] = None
+    strict: bool = True
+
+
+# Re-usable defaults
+GEO_PRODUCT_TYPES = NodeTypeProductTypes(
+    product_types=["pointcache", "model"],
+    default="pointcache"
+)
+FBX_PRODUCT_TYPES = NodeTypeProductTypes(
+    product_types=["fbx", "pointcache", "model"],
+    default="fbx"
+)
+FBX_ONLY_PRODUCT_TYPES = NodeTypeProductTypes(
+    product_types=["fbx"],
+    default="fbx"
+)
+USD_PRODUCT_TYPES = NodeTypeProductTypes(
+    product_types=["usd", "pointcache"],
+    default="usd"
+)
+COMP_PRODUCT_TYPES = NodeTypeProductTypes(
+    product_types=["imagesequence", "render"],
+    default="imagesequence"
+)
+REVIEW_PRODUCT_TYPES = NodeTypeProductTypes(
+    product_types=["review"],
+    default="review"
+)
+RENDER_PRODUCT_TYPES = NodeTypeProductTypes(
+    product_types=["render", "prerender"],
+    default="render"
+)
+GLTF_PRODUCT_TYPES = NodeTypeProductTypes(
+    product_types=["gltf"],
+    default="gltf"
+)
+
+
+class CreateHoudiniGeneric(plugin.HoudiniCreator):
+    """Generic creator to ingest arbitrary products"""
+
+    USE_DEFAULT_PRODUCT_TYPE = "__use_node_default__"
+    USE_DEFAULT_NODE_NAME = "__use_node_name__"
+
+    host_name = "houdini"
+
+    identifier = "io.ayon.creators.houdini.publish"
+    label = "Generic"
+    product_type = "generic"
+    icon = "male"
+    description = "Make any ROP node publishable."
+
+    render_target = "local_no_render"
+    default_variant = USE_DEFAULT_NODE_NAME
+    default_variants = ["Main", USE_DEFAULT_NODE_NAME]
+
+    # TODO: Move this to project settings
+    node_type_product_types: Dict[str, NodeTypeProductTypes] = {
```

Comment:
Why are not the types resolved during collection?

Reply:
These are supposed to be the "defaults" for new nodes to be created - not the ones currently set on the instance.

Reply:
Once `node_type_product_types` is moved to settings, this line should be replaced by applying/reading the settings?